cpu/kinetis/periph/i2c.c

```diff
@@ -45,474 +48,322 @@
 #endif
 
 /**
- * @brief Array holding one pre-initialized mutex for each I2C device
+ * @brief   Array of I2C module clock dividers
+ *
+ * The index in this array is the I2C_F setting number
  */
-static mutex_t locks[] =  {
-#if I2C_0_EN
-    [I2C_0] = MUTEX_INIT,
-#endif
-#if I2C_1_EN
-    [I2C_1] = MUTEX_INIT,
-#endif
-#if I2C_2_EN
-    [I2C_2] = MUTEX_INIT
-#endif
-#if I2C_3_EN
-    [I2C_3] = MUTEX_INIT
-#endif
+/*
+ * Divider values were copied from K22F reference manual
+ */
+static const uint16_t i2c_dividers[] = {
+      20,   22,   24,   26,   28,   30,   34,   40, /* 0x00 ~ 0x07 */
+      28,   32,   36,   40,   44,   48,   56,   68, /* 0x08 ~ 0x0f */
+      48,   56,   64,   72,   80,   88,  104,  128, /* 0x10 ~ 0x17 */
+      80,   96,  112,  128,  144,  160,  192,  240, /* 0x18 ~ 0x1f */
+     160,  192,  224,  256,  288,  320,  384,  480, /* 0x20 ~ 0x27 */
+     320,  384,  448,  512,  576,  640,  768,  960, /* 0x28 ~ 0x2f */
+     640,  768,  896, 1024, 1152, 1280, 1536, 1920, /* 0x30 ~ 0x37 */
+    1280, 1536, 1792, 2048, 2304, 2560, 3072, 3840, /* 0x38 ~ 0x3f */
 };
 
+static const i2c_conf_t i2c_config[I2C_NUMOF] = I2C_CONFIG;
```

Comment:
Why not directly defining this array of structs in the board periph_conf (like stm32, sam0, etc) instead of using the I2C_CONFIG macro ?

Reply:
I wanted to avoid unintentionally instancing this local array inside any other files, so I decided to put it as a define instead.

Reply:
Ok but then it's not consistent with the other configurations arrays (UART, SPI, PWN, ADC, etc).

Reply:
alright, will change this for now